src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/Tensor.cs

```diff
@@ -691,6 +691,55 @@ private Tensor<T> GetTriangle(int offset, bool upper)
         /// <param name="value">The new value to set at the specified position in this Tensor.</param>
         public abstract void SetValue(int index, T value);
 
+        /// <summary>
+        /// The type that implements enumerators for <see cref="Tensor{T}"/> instances.
+        /// </summary>
+        public struct Enumerator : IEnumerator<T>
+        {
+            private readonly Tensor<T> _tensor;
+            private int _index;
+
+            internal Enumerator(Tensor<T> tensor)
+            {
+                Debug.Assert(tensor != null);
+
+                _tensor = tensor;
+                _index = 0;
+                Current = default;
+            }
+
+            public T Current { get; private set; }
+
+            public bool MoveNext()
+            {
+                if (_index < _tensor.Length)
+                {
+                    Current = _tensor.GetValue(_index);
+                    ++_index;
+                    return true;
+                }
+                else
+                {
+                    Current = default;
+                    return false;
+                }
+            }
+
+            object? IEnumerator.Current => Current;
+
+            void IEnumerator.Reset()
+            {
+                this = default;
```

Comment:
This is going to overwrite _tensor and make subsequent use fail.  If the goal is to not implement it, it should throw a NotSupportedException.  Otherwise, it should reset _index and Current but leave _tensor alone.

Reply:
Yikes, not sure how I didn't notice that blatant null-reference bug. Fixed.